NeoML/src/TraditionalML/SMOptimizer.cpp

```diff
@@ -452,41 +496,41 @@ void CSMOptimizer::optimizePair( int i, int j, CArray<double>& alpha, CArray<dou
 				alpha[j] = -diff;
 			}
 		}
-		if(diff > weightI - weightJ) {
-			if(alpha[i] > weightI) {
-				alpha[i] = weightI;
-				alpha[j] = weightI - diff;
+		if(diff > C_i - C_j) {
+			if(alpha[i] > C_i) {
+				alpha[i] = C_i;
+				alpha[j] = C_i - diff;
 			}
 		} else {
-			if(alpha[j] > weightJ) {
-				alpha[j] = weightJ;
-				alpha[i] = weightJ + diff;
+			if(alpha[j] > C_j) {
+				alpha[j] = C_j;
+				alpha[i] = C_j + diff;
 			}
 		}
 	} else {
 		double quadCoef = QD[i] + QD[j] - 2 * Q_i[j];
-		if (quadCoef <= 0)
+		if( quadCoef <= 0 )
```

Comment:
Codestyle: figure brackets are missing

Reply:
I didn't touch this code, so let's better fix codestyle with separate PR.

Reply:
Agreed.